frontend/javascripts/test/puppeteer/dataset_rendering_helpers.ts

```diff
@@ -426,71 +436,62 @@ export async function withRetry(
   }
 }
 
-// Ava's recommendation for Typescript types
-// https://github.com/avajs/ava/blob/main/docs/recipes/typescript.md#typing-tcontext
-export const test = anyTest as TestFn<{
+// Define the test context type to be compatible with the test files
+export interface ScreenshotTestContext extends TestContext {
   browser: Browser;
   release?: () => void;
-}>;
+}
 
-export function setupBeforeEachAndAfterEach(semaphore?: Semaphore) {
-  test.beforeEach(async (t) => {
-    if (semaphore) {
-      t.context.release = await semaphore.acquire();
-    }
-    if (USE_LOCAL_CHROME) {
-      // Use this for connecting to local Chrome browser instance
-      t.context.browser = await puppeteer.launch({
-        args: HEADLESS
-          ? [
-              "--headless=false",
-              "--hide-scrollbars",
-              "--no-sandbox",
-              "--disable-setuid-sandbox",
-              "--disable-dev-shm-usage",
-              "--use-angle=gl-egl",
-            ]
-          : [],
-        headless: HEADLESS ? "new" : false, // use "new" to suppress warnings
-        dumpio: true,
-        executablePath: "/usr/bin/google-chrome", // this might need to be adapted to your local setup
-      });
-    } else {
-      checkBrowserstackCredentials();
-
-      const caps = {
-        browser: "chrome",
-        browser_version: "latest",
-        os: "os x",
-        os_version: "mojave",
-        name: t.title, // add test name to BrowserStack session
-        "browserstack.username": process.env.BROWSERSTACK_USERNAME,
-        "browserstack.accessKey": process.env.BROWSERSTACK_ACCESS_KEY,
-      };
-      const browser = await puppeteer.connect({
-        browserWSEndpoint: `ws://cdp.browserstack.com/puppeteer?caps=${encodeURIComponent(
-          JSON.stringify(caps),
-        )}`,
-      });
-      t.context.browser = browser;
-      console.log(`\nBrowserStack Session Id ${await getBrowserstackSessionId(browser)}\n`);
-    }
+export async function setupBeforeEach(context: ScreenshotTestContext, semaphore?: Semaphore) {
```

Comment:
This change is mostly identical just with different indentation. I split the method `setupBeforeEachAndAfterEach()` into two separate `setupBeforeEach()` and `setupAfterEach()` methods.